src/Gui/ReportView.cpp

```diff
@@ -170,6 +176,21 @@ void ReportHighlighter::highlightBlock (const QString & text)
 
         start = it->length;
     }
+    if(hasErrors || hasWarnings){

```

Comment:
Design-wise it's a bad idea to add this logic to the ReportHighlighter class because effectively this adds an unnecessary side-effect and reduces a lot its flexibility to be used in a different context as the global report window.

A better way would be to implement an observer class where to handle the visibility of the report view. The observer class should inherit from QObject and re-implement eventFilter. To the constructor you should pass the parent widget of the report view which is a QDockWidget (see below why). As a class member you ideally use a QPointer to keep the QDockWidget.

Reply:
What QEvent::type should I be looking for in eventFilter?  QEvent::User?  How do I determine which events are warnings or errors?

Reply:
ReportOutput uses the internal class CustomReportEvent with the type QEvent::User. So, in the eventFilter you check for the ReportOutput (which is the widget() of the QDockWidget) and QEvent::User. Then do a dynamic_cast to CustomReportEvent (also check that it's valid) which offers the method messageType().

This observer class is supposed to be instantiated inside MainWindow.cpp where the ReportOutput is instantiated. Then you must install the event filter to the application with: qApp->installEventFilter(reportoutputObserver);

Reply:
Okay thanks.  This is how I have the event filter installed in mainwindow.cpp already:

```
        ReportViewObserver* rvObserver = new ReportViewObserver((QDockWidget*)pcReport->parent());
        pcReport->installEventFilter(rvObserver);
```
I will change that to qApp->install... then work on the rest of it.  I can name the class ReportOutputObserver, too.  I think it describes its purpose better.

Reply:
A note for the future:
You should avoid doing the old C-style casts like (QDockWidget*)pcReport->parent() because they are not safe and often it's not clear if a static or reinterpret cast is done.

In C++ we have static, dynamic and reinterpret casts and here the best to use is either a dynamic cast or Qt's qobject_cast because they return a null pointer in case the cast is invalid.

For many years I avoid C-style and even use static cast for basic types like int, float, ...

No need to do this change as I will do it when merging the PR.